tidy3d/components/simulation.py

```diff
@@ -638,6 +639,97 @@ def _projection_monitors_homogeneous(cls, val, values):
 
         return val
 
+    @pydantic.validator("monitors", always=True)
+    def _projection_direction(cls, val, values):
+        """Warn if field projection observation points are behind surface projection monitors."""
+        # This validator is in simulation.py rather than monitor.py because volume monitors are
+        # eventually converted to their bounding surface projection monitors, in which case we
+        # do not want this validator to be triggered.
+
+        if val is None:
+            return val
+
+        with log as consolidated_logger:
+            for monitor_ind, monitor in enumerate(val):
+                if isinstance(monitor, AbstractFieldProjectionMonitor):
+                    if monitor.size.count(0.0) != 1:
+                        continue
+
+                    normal_dir = monitor.projection_surfaces[0].normal_dir
+                    normal_ind = monitor.size.index(0.0)
+
+                    projecting_backwards = False
+                    if isinstance(monitor, FieldProjectionAngleMonitor):
+                        r, theta, phi = np.meshgrid(
+                            monitor.proj_distance,
+                            monitor.theta,
+                            monitor.phi,
+                            indexing="ij",
+                        )
+                        x, y, z = Geometry.sph_2_car(r=r, theta=theta, phi=phi)
+                    elif isinstance(monitor, FieldProjectionKSpaceMonitor):
+                        uxs, uys, _ = np.meshgrid(
+                            monitor.ux,
+                            monitor.uy,
+                            monitor.proj_distance,
+                            indexing="ij",
+                        )
+                        theta, phi = monitor.kspace_2_sph(uxs, uys, monitor.proj_axis)
+                        x, y, z = Geometry.sph_2_car(r=monitor.proj_distance, theta=theta, phi=phi)
+                    else:
+                        pts = monitor.unpop_axis(
+                            monitor.proj_distance, (monitor.x, monitor.y), axis=normal_ind
+                        )
+                        x, y, z = pts
+
+                    center = np.array(monitor.center) - np.array(monitor.local_origin)
+                    pts = [np.array(i) for i in [x, y, z]]
+                    normal_displacement = pts[normal_ind] - center[normal_ind]
```

Comment:
Is doing it this way easier (and also is it equivalent?) to checking if theta < 0?

Reply:
The angle defined within the monitor is relative to the local origin, so we'd need to do some conversions anyway because theta < 0 in the local system may still not be "backwards" in the global system, relative to the monitor's position, depending on `proj_distance`. So I decided to always convert to the global system in Cartesian coords and treat all three cases in the same way - not sure if it's the best way

Reply:
another way could be to always convert to spherical coords in the global system and then check if the global theta < 0

Reply:
Oh ok I see.